robot/web/web.go

```diff
@@ -179,12 +179,12 @@ func (svc *webService) Start(ctx context.Context, o weboptions.Options) error {
 }
 
 // RunWeb starts the web server on the robot with web options and blocks until we cancel the context.
-func RunWeb(ctx context.Context, r robot.LocalRobot, o weboptions.Options, logger logging.Logger) (err error) {
+func RunWeb(ctx context.Context, r robot.LocalRobot, o weboptions.Options, shutdownLogger logging.Logger) (err error) {
 	defer func() {
 		if err != nil {
 			err = utils.FilterOutError(err, context.Canceled)
 			if err != nil {
-				logger.Errorw("error running web", "error", err)
+				shutdownLogger.Errorw("error running web", "error", err)
```

Comment:
It wasn't obvious to me this change was right. But it is.

Can we undo this weird defer thing? And just assume `StartWeb` is not being weird and panicing? And maybe change the message to be understandable(-ish) by a user: `Error running RPC server. Shutting down.`

At least I think startweb's primary purpose is to allow for (g)RPC invocations?